syntax = "proto3";

package yandex.cloud.audit.loadbalancer;

import "google/protobuf/timestamp.proto";
import "google/rpc/status.proto";
import "yandex/cloud/audit/common.proto";
import "yandex/cloud/validation.proto";
import "yandex/cloud/loadbalancer/v1/target_group.proto";

option go_package = "a.yandex-team.ru/cloud/bitbucket/public-api/yandex/cloud/audit/loadbalancer;loadbalancer";
option java_package = "yandex.cloud.api.audit.loadbalancer";
option java_outer_classname = "NetworkTargetGroupEvents";

message CreateTargetGroup {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string target_group_id = 1 [(required) = true];
        string target_group_name = 2 [(required) = true];
        string region_id = 3 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.Target targets = 4;
    }
}

message UpdateTargetGroup {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string target_group_id = 1 [(required) = true];
        string target_group_name = 2 [(required) = true];
        string region_id = 3 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.Target targets = 4;
    }
}

message DeleteTargetGroup {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string target_group_id = 1 [(required) = true];
        string target_group_name = 2 [(required) = true];
    }
}

message AddTargetGroupTargets {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string target_group_id = 1 [(required) = true];
        string target_group_name = 2 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.Target targets = 3;
    }
}

message RemoveTargetGroupTargets {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string target_group_id = 1 [(required) = true];
        string target_group_name = 2 [(required) = true];
        repeated yandex.cloud.loadbalancer.v1.Target targets = 3;
    }
}
